rust/core-lib/src/tabs.rs

```diff
@@ -566,11 +568,16 @@ impl CoreState {
 
     fn finalize_new_views(&mut self) {
         let to_start = mem::replace(&mut self.pending_views, Vec::new());
-        to_start.iter().for_each(|(id, config)| {
-            let modified = self.detect_whitespace(*id, config);
-            let config = modified.as_ref().unwrap_or(config);
-            let mut edit_ctx = self.make_context(*id).unwrap();
-            edit_ctx.finish_init(&config);
+        to_start.iter().for_each(|(id, config, is_existing_file)| {
+            if *is_existing_file {
```

Comment:
instead of passing through `is_existing_file`, we could just do,

```rust
let has_contents = self.editors.get(&buffer_id)
    .map(|e| !e.get_buffer().is_empty())
    .unwrap_or_default();
if has_contents {
    // ...
} else {
    // ...
}
```

This might be easier? It means less state to worry about.

Reply:
Yes. So that only when open an non-empty file whitespace will be checked.